mathematical-companion/main.tex

```diff
@@ -0,0 +1,1300 @@
+
+The run of 8 consecutive roots is the reason that this code has distance 9, though
+the reason is not one we can casually state.\footnote{It can be easily
+found online, e.g.~on the Wikipedia page for BCH codes.}
+
+Now, recall $G$ has degree 2, so all of its roots have degree at most 2, or
+equivalently, it can be factored in $\fttwo$ into factors which are all linear
+or quadratic. Specifically:
+\begin{align*}
+G(x) = (x &+ \vc{T})(x + \vc{S})(x + \vc{C}) \\
+    &\times (x^2 + \vc{Z}x + \vc{Y})(x^2 + \vc{R}x + \vc{9})
+    (x^2 + \vc{2}x + \vc{K})(x^2 + \vc{W}x + \vc{X})(x^2 + \vc{L}x + \vc{A})
+\end{align*}
+
+Recall further that our codewords are defined by their membership in a specific
+equivalence class modulo $G$, the equivalence class of \vc{SECRETSHARE32}. By
+the Chinese Remainder Theorem, this equivalence class is also characterized by
+sets of equivalence classes modulo these factors. In particular, for a given $v(x)$,
+\begin{align}
+v(x) &\equiv \vc{2W} \label{eq:quickstart}
+  &\mod (x + \vc{S})(x + \vc{T})
+  &= (x - \beta^{84})(x - \beta^{81}) \\
+v(x) &\equiv \vc{LD}
+  &\mod (x^2 + \vc{Z}x + \vc{Y})
+  &= (x - \beta^{83})(x - \beta^{52}) \\
+v(x) &\equiv \vc{XK}
+  &\mod (x^2 + \vc{2}x + \vc{K})
+  &= (x - \beta^{82})(x - \beta^{20}) \\
+v(x) &\equiv \vc{9X}
+  &\mod (x^2 + \vc{W}x + \vc{X})
+  &= (x - \beta^{80})(x - \beta^{49}) \\
+v(x) &\equiv \vc{LT}
+  &\mod (x^2 + \vc{L}x + \vc{A})
+  &= (x - \beta^{79})(x - \beta^{17}) \\
+v(x) &\equiv \vc{WU}
+  &\mod (x + \vc{C})(x + \vc{T})
+  &= (x - \beta^{78})(x - \beta^{81}) \\
+v(x) &\equiv \vc{UM} \label{eq:quickend}
+  &\mod (x^2 + \vc{R}x + \vc{9})
+  &= (x - \beta^{77})(x - \beta^{46})
+\end{align}
+if and only if $v(x)\equiv\vc{SECRETSHARE32}$ modulo $G$.
+
+In fact, we have ordered these checks so that if the user checks each one of
+them, in order, the CRT will fix the equivalence class of her codeword modulo
+a polynomial $\hat{G}$ with progressively many consecutive roots of $\beta$,
+so she will be guaranteed to detect progressively many errors.
+
+This is the intuition behind the \textbf{quickcheck} method of verifying the
+checksum. Rather than, say, doing a full checksum worksheet every year to verify
+their codeword module $G$, the user can instead verify each of
+Equations~\eqref{eq:quickstart} through~\eqref{eq:quickend}, doing one check
+ever month. Once the final check~\eqref{eq:quickend} is done, the user starts
+back over with~\eqref{eq:quickstart}.
+
+You may notice that the factor $(x + \vc{S}) = (x - \beta^{81})$ appears twice in
+this list. For a user doing all the checks, its second appearance is strictly
+redundant. But by including it, we ensure that no matter where in the list the
+user is starting from, they consistently accumulate consecutive roots, so that
+if their data isn't corrupted between individual checks, they gain progressive
+error detection ability.
+
+Another reason to include $(x+\vc{S})$ twice is to make all the quickchecks look
```

Comment:
The most important reason is that a quickcheck detects any other types of errors with 99.9% probability (1023/ 1024).  If we only did a linear check that would reduce the detection probability to 31/32.